src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -84,6 +101,41 @@ public TransportService(Settings settings, Transport transport, ThreadPool threa
         super(settings);
         this.transport = transport;
         this.threadPool = threadPool;
+        this.tracerLogInclude = settings.getAsArray(SETTING_TRACE_LOG_INCLUDE, Strings.EMPTY_ARRAY, true);
+        this.tracelLogExclude = settings.getAsArray(SETTING_TRACE_LOG_EXCLUDE, new String[]{"internal:discovery/zen/fd*"}, true);
+        tracerLog = Loggers.getLogger(logger, ".tracer");
+    }
+
+    // These need to be optional as they don't exist in the context of a transport client
+    @Inject(optional = true)
+    public void setDynamicSettings(NodeSettingsService nodeSettingsService, @ClusterDynamicSettings DynamicSettings dynamicSettings) {
+        dynamicSettings.addDynamicSettings(SETTING_TRACE_LOG_INCLUDE, SETTING_TRACE_LOG_INCLUDE + ".*");
+        dynamicSettings.addDynamicSettings(SETTING_TRACE_LOG_EXCLUDE, SETTING_TRACE_LOG_EXCLUDE + ".*");
+        nodeSettingsService.addListener(settingsListener);
+    }
+
+
+    class ApplySettings implements NodeSettingsService.Listener {
+        @Override
+        public void onRefreshSettings(Settings settings) {
+            String[] newTracerLogInclude = settings.getAsArray(SETTING_TRACE_LOG_INCLUDE, TransportService.this.tracerLogInclude, true);
+            String[] newTracerLogExclude = settings.getAsArray(SETTING_TRACE_LOG_EXCLUDE, TransportService.this.tracelLogExclude, true);
+            if (newTracerLogInclude == TransportService.this.tracerLogInclude && newTracerLogExclude == TransportService.this.tracelLogExclude) {
+                return;
+            }
+            if (Strings.arrayToCommaDelimitedString(newTracerLogInclude).equals(Strings.arrayToCommaDelimitedString(TransportService.this.tracerLogInclude)) &&
```

Comment:
could you use `Arrays.equals` here? Both variants do not take order of the array into account... but no need to overoptimize here